pkg/ipcache/kvstore.go

```diff
@@ -382,10 +382,12 @@ var (
 func InitIPIdentityWatcher() {
 	setupIPIdentityWatcher.Do(func() {
 		globalMap = newKVReferenceCounter(kvstoreImplementation{})
-		log.Info("Starting IP identity watcher")
-		watcher = NewIPIdentityWatcher(kvstore.Client())
-		close(initialized)
-		go watcher.Watch()
+		go func() {
+			log.Info("Starting IP identity watcher")
+			watcher = NewIPIdentityWatcher(kvstore.Client())
```

Comment:
This is not enough, you need a lock to protect the `watcher` variable assignment as it can now be accessed in parallel. This was the reason it was moved out.